scripts/proto-gen.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
```

Comment:
should we bash explicitly to avoid the bash/dash/ash ambiguity

Reply:
Oops, I missed this comment. I think it should be fine—we're not relying on anything that POSIX doesn't require (so the ambiguity should be safe).

Reply:
ah yeah, the pipefail doesn't work in dash, (alas)